### DISCUSSION TITLE
"Show me some code" example

### DISCUSSION THREAD

**Discussion by jmafc** [hylo-lang/hylo#841] [Q&A] [answered]
I've built `valc` on Debian 12 and ran its tests. Before going through the language tour, I thought I'd try the "Enough, show me some code" example on the home page. I was surprised because `valc` pointed out that type `String` has no members `count` or `append`.  Looking at `Library/Val/Core/String.val` appears to confirm that, unless Swift provides that and some command line argument will fix the problem.  So unless I'm wrong, it would seem that the "some code" is what Val is expected to do in the future. I have no problem with that but, if so, I would recommend adding some indication that the example does not work with the current compiler.

Also, I noticed that the `-l <name>` command line option is described as linking `the generated crate(s)`. Is that intentional? I'm asking because I thought that `crates` was a Rust-specific term and I haven't seen mention that Val is adopting that convention.

**Comment by kyouko-taiga** [comment, marked as answer]
You guessed right. The code is a just an example of what we'll get in the future. We'll add a note, as you suggest.

As for the "crate", I don't know how it slipped there. Could you open an issue on the compiler's repo?

**Reply by jmafc** [reply]
Thanks for the clarification.

OK, will do.
